components/ChannelDetails/StakeDialog.tsx

```diff
@@ -0,0 +1,80 @@
+import { Button } from '@/components/ui/button';
+import { Dialog, DialogContent } from '@/components/ui/dialog';
+import { Drawer, DrawerContent } from '@/components/ui/drawer';
+import { Input } from '@/components/ui/input';
+import formatNumber from '@/lib/formatNumber';
+import { cn } from '@/lib/utils';
+import { useUi } from '@/providers/UiProvider';
+import Image from 'next/image';
+import { useState } from 'react';
+import Tabs from './StakeTabs';
+
+function StakeDialog({ balance = 0, disabled = true }) {
+  const { isMobile } = useUi();
+  const [isOpen, setIsOpen] = useState(false);
+
+  return (
+    <>
+      <div className='text-center'>
+        <Button
+          disabled={disabled}
+          onClick={() => setIsOpen(true)}
+          className="h-auto rounded-full px-9 py-4 text-base font-normal">
+          STAKE NOTES
+        </Button>
+        <p className='mt-2 text-sm font-semibold'>
+          <span className='text-sm font-normal text-grey'>Staked: </span>
+          {formatNumber(balance)} NOTES
+        </p>
+      </div>
+
+      <Dialog open={!isMobile && isOpen} onOpenChange={setIsOpen}>
+        <DialogContent className='max-w-[709px] !rounded-3xl p-8'>
+          <Body balance={balance} className='gap-8' />
+        </DialogContent>
+      </Dialog>
+
+      <Drawer open={isMobile && isOpen} onOpenChange={setIsOpen}>
+        <DrawerContent>
+          <Body balance={balance} className='gap-6 p-6' />
+        </DrawerContent>
+      </Drawer>
+    </>
+  );
+}
+
+const tabs = [{ label: 'Stake', value: true }, { label: 'Unstake', value: false }];
+
+function Body({ balance = 0, className = '' }) {
```

Comment:
each file should do EXACTLY one thing
- actual: this file is defining BOTH `Body` and `StakeDialog`
- expected: separate files for each component definition